lib/Async/Registry/promise.h

```diff
@@ -44,37 +44,55 @@ auto inspect(Inspector& f, Thread& x) {
                             f.field("id", fmt::format("{}", x.id)));
 }
 
+struct SourceLocation {
+  const std::string file_name;
+  const std::string function_name;
```

Comment:
I think it is safe to assume that `file_name` and `function_name` will point to a static `const char*` buffer in the data segment. We don't have to copy those onto the heap. Instead of `std::string` we could use `std::string_view`. 

However, we have to keep an eye on this, because nowhere on cppreference it was written explicitly that those strings are static.